bluemira/base/parameter_frame/_frame.py

```diff
@@ -102,7 +102,9 @@ def update_from_dict(self, new_values: ParamDictT):
             value_type = _validate_parameter_field(key, self._types[key])
             new_param = Parameter(name=key, **value, _value_types=value_type)
             param.set_value(
-                new_param.value if param.unit == "" else new_param.value_as(param.unit),
+                new_param.value
+                if param.unit == "" or new_param.value is None
+                else new_param.value_as(param.unit),
```

Comment:
We've got this logic in two places, I can see the other instance was already not matching this. I'm trying to think of the best way to refactor it out.

Maybe the `None` logic could go in `Parameter.set_value`? So, if the `Parameter`'s value is `None`, just return `None`, regardless of the unit (or maybe we should enforce that the units are compatible?). I can't remember what the `param.unit == ""` is for. I guess to allow the unit not to be specified in the dict if we just want to update the value?

Thoughts? I think if we are to not refactor the logic, we should at least leave a comment in both places saying that these should be consistent.

Reply:
the `unit == ""` will avoid all unit conversions on bools str etc. 

On the refactor yeah you're right. I'll have a play with what you suggested and see what comes out

Reply:
I've opted for a refactor within `ParameterFrame` because of setting long name and description which was already duplicated as you mentioned. 

I havent moved the None check into `Parameter` because we convert units before we set the value.
The other option is that `param.value_as` should only convert on that conditional. 
As I'm writing this I am thinking I could do that too.

Thoughts?

Reply:
I think keeping the logic within `ParameterFrame`, as you have now, is fine. As you say, we need to do something about `param.value_as` when the value is `None`. Currently, you just get an error. Options are:

1. Just return `None` whatever the unit is.
2. Validate that the unit is compatible, before returning `None`, else throw.
3. Just throw an error (unless trying to retrieve the value as its existing unit?).

I'm struggling with which is best 😅. I'm leaning towards 2, as I think it's good for a user to know that they're trying to do something wrong. The value may not be `None` if they run the same code with different settings next time. Better to flag the bug early, I think.

Reply:
I agree, done the second option